comp/dogstatsd/packets/buffer.go

```diff
@@ -67,15 +71,19 @@ func (pb *Buffer) Append(packet *Packet) {
 func (pb *Buffer) flush() {
 	if len(pb.packets) > 0 {
 		t1 := time.Now()
+
+		TelemetryTrackPackets(pb.packets, pb.listenerID)
+		tlmBufferSizeBytes.Add(-float64(pb.packets.SizeInBytes()), pb.listenerID)
```

Comment:
The metrics are gauges that are sampled once every 15s, but change many times per second. This is ends up very as a very low sample rate, and doesn't capture at all what happens in between. Worse still if the load also happens at 15s cadence - chances are we'll never even see a random sample of it. Would it make sense to track this as a histogram, or at least min/max/avg set of values?

Reply:
That is a very good point. But I think simple gauges are enough since we mostly care about sustained load scenarios (when the channel is full), and short spikes don't matter much to us (the client should just have higher timeouts)